mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -30,3 +35,26 @@ def csv_stdlib(fp):
         })
     rows = [row for row in reader]
     return columns, rows
+
+
+def _set_dialect_quote_attrs(dialect, data):
+    """Set quote-related dialect attributes based on up to 2kb of csv data.
+
+    The regular expressions search for things that look like the beginning of
+    a list, wrapped in a quotation mark that is not dialect.quotechar, with
+    list items wrapped in dialect.quotechar and seperated by commas.
+
+    Example matches include:
+        "['1', '2', '3'         for quotechar == '
+        '{"a", "b", "c"         for quotechar == "
+    """
+    if dialect.quotechar == '"':
+        if re.search('\'[[({]\".+\",', data):
+            dialect.quotechar = "'"
+        if re.search('\'\'\'[[({]\".+\",', data):
```

Comment:
When writing regexes avoid escaping characters as much as possible.
Use raw strings, `r'...'` and swap double and single quotes as need be.
`r"'''[[({]\".+\","`